File: java/classes/ClassEncapsulation.java
class Account {
  boolean isActive = false; // public by default
  public String login = "";
  public String email = "";
  public int id = 0;

  protected int[] connections = new int[10];
  private int connectionsNumber = 0;

  public void display() {
    System.out.println("ID: " + id + "\n"
      + "Login: " + login + "\n"
      + "E-mail: " + email + "\n"
      + "Is active: " + isActive + "\n"
      + "Has connections: " + hasConnections()
    );
  };

  public boolean hasConnections() {
    return (connectionsNumber > 0);
  };
  public boolean addConnection(int connectedAccountId) {
    if (connectionsNumber == 10)
      return false;

    connections[connectionsNumber] = connectedAccountId;
    connectionsNumber++;

    return true;
  }
}

class SocialMediaAccount extends Account {
  public boolean isFriend(int checkingAccountId) {
    for(int connectionAccountId : connections) {
      if (checkingAccountId == connectionAccountId)
        return true;
    }

    return false;
  }
}

class ClassEncapsulation {
  public static void main(String[] args) {
    SocialMediaAccount timothy = new SocialMediaAccount();

    timothy.login = "tim";
    timothy.email = "timothy.muppetone@gmail.com";
    timothy.isActive = true;

    timothy.display();
    System.out.println();

    timothy.addConnection(100);
    System.out.println("Has connections: " + timothy.hasConnections());
    System.out.println("Is ID 100 a friend? " + timothy.isFriend(100));
    System.out.println("Is ID 100 a friend? " + timothy.isFriend(200));
  }
}

File: c#/classes/ClassEncapsulation.cs
class Account {
  bool IsActive = false; // private by default
  public string Login = "";
  public string Email = "";
  public int Id = 0;

  public const int ConnectionsMaxNumber = 10;
  protected int[] Connections = new int[ConnectionsMaxNumber];
  private int ConnectionsNumber = 0;

  public void Display() {
    System.Console.WriteLine("ID: " + Id + "\n"
      + "Login: " + Login + "\n"
      + "E-mail: " + Email + "\n"
      + "Is active: " + IsActive + "\n"
      + "Has connections: " + HasConnections()
    );
  }

  public bool HasConnections() {
    return (ConnectionsNumber > 0);
  }
  public bool AddConnection(int connectedAccountId) {
    if (ConnectionsNumber == ConnectionsMaxNumber)
      return false;

    Connections[ConnectionsNumber] = connectedAccountId;
    ConnectionsNumber++;

    return true;
  }
}

class SocialMediaAccount : Account {
  public bool IsFriend(int checkingAccountId) {
    foreach(int connectionAccountId in Connections) {
      if (checkingAccountId == connectionAccountId)
        return true;
    }

    return false;
  }
}

class ClassEncapsulation {
  public static void Main(string[] args) {
    SocialMediaAccount timothy = new SocialMediaAccount();

    timothy.Login = "tim";
    timothy.Email = "timothy.muppetone@gmail.com";
    // timothy.IsActive = true; // cannot access private component

    timothy.Display();
    System.Console.WriteLine();

    timothy.AddConnection(100);
    System.Console.WriteLine("Has connections: " + timothy.HasConnections());
    System.Console.WriteLine("Is ID 100 a friend? " + timothy.IsFriend(100));
    System.Console.WriteLine("Is ID 100 a friend? " + timothy.IsFriend(200));
  }
}
